### PR TITLE
docs: added pub/sub background study doc

### PR DESCRIPTION
**Pull request by Mossaka**

### CODE REVIEW

File: docs/pubsub-background.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+[[capability]]
+name = "pubsub.mqtt"
+[[capability.config]]
+url = "tcp://localhost:1883"
+qos = 1
+topic = "group"
+delivery = "push"
+on_message = "on_message"
+```
```
Comment (danbugs) [inline docs/pubsub-background.md:122]: Did you mean to keep the design stuff here? I'd probably remove it to leave discussions on the design to of the interface pubsub re-design proposal.

File: docs/pubsub-background.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# Pub/Sub Study
+
+This document describes the design of the Pub/Sub capability.
```
Comment (danbugs) [inline docs/pubsub-background.md:3]:
```suggestion
This document describes some background research that inspired the design of the Pub/Sub capability.
```

File: docs/pubsub-background.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+## Background
```
Comment (danbugs) [inline docs/pubsub-background.md:5]:
```suggestion
## Definition
```

**Review by danbugs — CHANGES_REQUESTED** [review]

File: docs/pubsub-background.md (hunk 1 above)
Comment (Mossaka) [reply docs/pubsub-background.md:122]: Nope. I can remove them

**Pull request merged**